Tools/cases_generator/stack.py

```diff
@@ -370,3 +404,58 @@ def stacks(inst: Instruction | PseudoInstruction) -> Iterator[StackEffect]:
                 local = Local.unused(var)
             stack.push(local)
     return stack
+
+@dataclass
+class Storage:
+
+    stack: Stack
+    outputs: list[Local]
+
+    def _push_defined_locals(self) -> None:
+        while self.outputs:
+            out = self.outputs[0]
+            if out.defined:
+                self.stack.push(out)
+                self.outputs.pop(0)
+            else:
+                break
+        undefined = ""
+        for out in self.outputs:
+            if out.defined:
```

Comment:
If the first `out.defined` holds, then we don't really have anything "before" (unless this case is not possible?)